### PR TITLE
[PORT] - Ports TG's Chat announcement style

### PR DESCRIPTION
**Pull request by cowbot92**
# Document the changes in your pull request

Adds a whole lot of stuff, mostly the stuff for the TGUI and chat text for the """new""" announcement style & some backend stuff for achievements when I port that (god save me)

# Why is this good for the game?

it looks so cool 

image

# Testing
Communications console report looks like this:
image

Admin Panel updated w/ TGUI:
image

Example of what you can send & the options to choose from:
image
image

# Changelog

:cl:  

tweak: Ports updated TG announcements & TGUI

/:cl:

### CODE REVIEW

File: code/__DEFINES/achievements.dm [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+#define ACHIEVEMENT_SCORE "score"
+
+/// preferences for the sound played when unlocking an achievement
+#define CHEEVO_SOUND_TADA "Tada Fanfare"
+#define CHEEVO_SOUND_JINGLE "Beeps Jingle"
+#define CHEEVO_SOUND_PING "Success Ping"
+#define CHEEVO_SOUND_OFF "Disabled"
+
+//Misc Medal hub IDs
+#define MEDAL_METEOR "Your Life Before Your Eyes"
```
Comment (ToasterBiome) [inline code/__DEFINES/achievements.dm]: Why are these here?

File: code/__DEFINES/shuttles.dm [hunk 2]
```diff
@@ -87,3 +87,10 @@
 ///Check for arena shuttle, if the bubblegum has died this round
 GLOBAL_VAR_INIT(bubblegum_dead, FALSE)
 
+// Alert level related for new thing im porting --- cowbot93
+#define ALERT_COEFF_AUTOEVAC_NORMAL 2.5
```
Comment (ToasterBiome) [inline code/__DEFINES/shuttles.dm]: Check to ensure these are even the same ratios that we actually use

File: code/__DEFINES/tts.dm [hunk 3]
```diff
@@ -0,0 +1,10 @@
+///TTS preference is disbaled entirely, no sound will be played.
```
Comment (ToasterBiome) [inline code/__DEFINES/tts.dm]: why are TTS prefs being added here?

File: code/_globalvars/achievements.dm [hunk 4]
```diff
@@ -0,0 +1,9 @@
+GLOBAL_LIST_EMPTY(commendations_achievments)
```
Comment (ToasterBiome) [inline code/_globalvars/achievements.dm]: This isnt spelled right

File: code/_globalvars/achievements.dm [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+///A list of the current achievement categories supported by the UI and checked by the achievement unit test
```
Comment (ToasterBiome) [inline code/_globalvars/achievements.dm]: why are you porting achievement stuff

File: code/modules/admin/verbs/commandreport.dm [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+/// Preset central command names to chose from for centcom reports.
+#define CENTCOM_PRESET "Central Command"
+#define SYNDICATE_PRESET "The Syndicate"
+#define WIZARD_PRESET "The Wizard Federation"
+#define CUSTOM_PRESET "Custom Command Name"
+
+/// Verb to change the global command name.
+/client/proc/cmd_change_command_name()
+	set category = "Admin.Events"
```
Comment (ToasterBiome) [inline code/modules/admin/verbs/commandreport.dm]: should stay in the same category as the old command

File: code/modules/admin/verbs/commandreport.dm [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+	var/input = input(usr, "Please input a new name for Central Command.", "What?", "") as text|null
+	if(!input)
+		return
+	change_command_name(input)
+	message_admins("[key_name_admin(src)] has changed Central Command's name to [input]")
+	log_admin("[key_name(src)] has changed the Central Command name to: [input]")
+
+/// Verb to open the create command report window and send command reports.
+/client/proc/cmd_admin_create_centcom_report()
+	set category = "Admin.Events"
```
Comment (ToasterBiome) [inline code/modules/admin/verbs/commandreport.dm]: should stay in the same category as the old command

File: code/modules/client/client_preferences_sounds.dm [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+	savefile_identifier = PREFERENCE_PLAYER
+
+/// Controls hearing announcement sounds
+/datum/preference/toggle/sound_announcements
+	category = PREFERENCE_CATEGORY_GAME_PREFERENCES
+	savefile_key = "sound_announcements"
+	savefile_identifier = PREFERENCE_PLAYER
+
+/// Controls hearing the combat mode toggle sound
+/datum/preference/toggle/sound_combatmode
```
Comment (ToasterBiome) [inline code/modules/client/client_preferences_sounds.dm]: Nope

File: code/modules/client/client_preferences_sounds.dm [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+	savefile_identifier = PREFERENCE_PLAYER
+
+/// Controls hearing round end sounds
+/datum/preference/toggle/sound_endofround
+	category = PREFERENCE_CATEGORY_GAME_PREFERENCES
+	savefile_key = "sound_endofround"
+	savefile_identifier = PREFERENCE_PLAYER
+
+/// Controls hearing instruments
+/datum/preference/toggle/sound_instruments
```
Comment (ToasterBiome) [inline code/modules/client/client_preferences_sounds.dm]: If you're porting all these prefs you need to
1) remove the old prefs
2) make the new prefs be used

File: code/modules/mob/living/silicon/ai/decentralized/management/ai_controlpanel.dm [hunk 10]
```diff
@@ -320,7 +320,7 @@ GLOBAL_VAR_INIT(ai_control_code, random_nukecode(6))
 				one_time_password_used = TRUE
 				var/msg = "<h4>Warning!</h4><br>We have detected usage of the AI Control Code for unlocking a console at coordinates ([src.x], [src.y], [src.z]) by [usr.name]. Please verify that this is correct. Be aware we have cancelled the current control code.<br>\
 				If needed a new code can be printed at a communications console."
-				priority_announce(msg, sender_override = "Central Cyber Security Update", has_important_message = TRUE, sanitize = FALSE)
+				priority_announce(msg, sender_override = "Central Cyber Security Update", has_important_message = TRUE)
```
Comment (ToasterBiome) [inline code/modules/mob/living/silicon/ai/decentralized/management/ai_controlpanel.dm:323]: Does this still work? sanitize might be off for a reason.

File: code/modules/security_levels/security_level_datums.dm [hunk 11]
```diff
@@ -0,0 +1,92 @@
+/**
+ * Security levels
+ *
+ * These are used by the security level subsystem. Each one of these represents a security level that a player can set.
+ *
+ * Base type is abstract
+ */
+
+/datum/security_level
```
Comment (ToasterBiome) [inline code/modules/security_levels/security_level_datums.dm:9]: Missing epsilon and gamma levels (idc if they work without them, find out why it works without them, because it shouldnt)

File: code/modules/shuttle/emergency.dm [hunk 12]
```diff
@@ -452,7 +452,7 @@
 					destination_dock = "emergency_syndicate"
 					minor_announce("Corruption detected in \
 						shuttle navigation protocols. Please contact your \
-						supervisor.", "SYSTEM ERROR:", alert=TRUE, custom_alert_sound = 'sound/misc/announce2.ogg')
+						supervisor.", "SYSTEM ERROR:", alert=TRUE)
```
Comment (ToasterBiome) [inline code/modules/shuttle/emergency.dm:455]: is this sound playing moved somewhere else?

File: code/modules/security_levels/security_levels.dm [hunk 13]
```diff
@@ -53,7 +53,8 @@ GLOBAL_VAR_INIT(security_level, SEC_LEVEL_GREEN)
 
 			if(SEC_LEVEL_RED)
 				if(GLOB.security_level < SEC_LEVEL_RED)
-					minor_announce(CONFIG_GET(string/alert_red_upto), "Attention! Code red!", TRUE, custom_alert_sound = 'sound/misc/notice4.ogg')
+					minor_announce(CONFIG_GET(string/alert_red_upto), "Attention! Code red!", TRUE)
+					sound_to_playing_players('sound/misc/notice4.ogg')
```
Comment (ToasterBiome) [inline code/modules/security_levels/security_levels.dm:59]: why isnt this custom sound in a datum/security_level entry so we can set it manually per level?

**Review by ToasterBiome — CHANGES_REQUESTED** [review]

File: code/__DEFINES/achievements.dm (hunk 1 above)
Comment (cowbot92) [reply code/__DEFINES/achievements.dm]: gonna remove this

File: code/__DEFINES/shuttles.dm (hunk 2 above)
Comment (cowbot92) [reply code/__DEFINES/shuttles.dm]: adjusted to same ratios in next push

File: code/__DEFINES/tts.dm (hunk 3 above)
Comment (cowbot92) [reply code/__DEFINES/tts.dm]: removing this

File: code/_globalvars/achievements.dm (hunk 4 above)
Comment (cowbot92) [reply code/_globalvars/achievements.dm]: corrected

File: code/modules/client/client_preferences_sounds.dm (hunk 9 above)
Comment (cowbot92) [reply code/modules/client/client_preferences_sounds.dm]: removing this

File: code/modules/mob/living/silicon/ai/decentralized/management/ai_controlpanel.dm (hunk 10 above)
Comment (cowbot92) [reply code/modules/mob/living/silicon/ai/decentralized/management/ai_controlpanel.dm:323]: will double check but it worked when I tested it

File: code/modules/security_levels/security_level_datums.dm (hunk 11 above)
Comment (cowbot92) [reply code/modules/security_levels/security_level_datums.dm:9]: ugh fine

File: code/modules/shuttle/emergency.dm (hunk 12 above)
Comment (cowbot92) [reply code/modules/shuttle/emergency.dm:455]: readded this back

File: code/modules/security_levels/security_levels.dm (hunk 13 above)
Comment (cowbot92) [reply code/modules/security_levels/security_levels.dm:59]: because it only ever plays when red alert is set

File: code/_globalvars/achievements.dm (hunk 5 above)
Comment (cowbot92) [reply code/_globalvars/achievements.dm]: removed this

File: code/__DEFINES/tts.dm (hunk 3 above)
Comment (cowbot92) [reply code/__DEFINES/tts.dm]: removed

File: code/modules/security_levels/security_level_datums.dm (hunk 11 above)
Comment (cowbot92) [reply code/modules/security_levels/security_level_datums.dm:9]: Readded them and also i found out why it works without them (It didnt lol!)

File: code/modules/security_levels/security_levels.dm (hunk 13 above)
Comment (cowbot92) [reply code/modules/security_levels/security_levels.dm:59]: wait im stupid i think I see what you mean

File: code/_globalvars/achievements.dm
Comment (ToasterBiome) [inline code/_globalvars/achievements.dm:1]: empty file

File: code/modules/admin/verbs/randomverbs.dm [hunk 14]
```diff
@@ -1598,3 +1557,179 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+
+/// Preset central command names to chose from for centcom reports.
+#define CENTCOM_PRESET "Central Command"
+#define SYNDICATE_PRESET "The Syndicate"
+#define WIZARD_PRESET "The Wizard Federation"
+#define CUSTOM_PRESET "Custom Command Name"
+
+/// Verb to change the global command name.
+/client/proc/cmd_change_command_name()
+	set category = "Admin.Events"
```
Comment (ToasterBiome) [inline code/modules/admin/verbs/randomverbs.dm]: i didnt mean move it to this file i meant the set category part should be "Admin.Round Interaction" like it used to be

**Review by ToasterBiome — CHANGES_REQUESTED** [review]

**Comment by cowbot92** [comment]
I am unable to get the sounds to work on the announcements (minus the alert level changes) and I don't think I'll have the energy or time going forward to fix it.

**Comment by cowbot92** [comment]
fuck it we ball

**Review by ToasterBiome — APPROVED** [review]

**Comment by Beeper-Skeeper** [comment]
I may be day-banned on the discord but that won't stop me from saying "HOLY SHIT THIS SHIT COOL"
